src/client/clientmap.cpp

```diff
@@ -553,17 +567,24 @@ void ClientMap::touchMapBlocks()
 			v3f mesh_sphere_center = intToFloat(block->getPosRelative(), BS)
 					+ block->mesh->getBoundingSphereCenter();
 			f32 mesh_sphere_radius = block->mesh->getBoundingRadius();
+
+			float range = mesh_sphere_center.getDistanceFrom(cam_pos_f);
+			if (range > loaded_range)
+				loaded_range = range;
+
 			// First, perform a simple distance check.
 			if (!m_control.range_all &&
-				mesh_sphere_center.getDistanceFrom(intToFloat(cam_pos_nodes, BS)) >
-					m_control.wanted_range * BS + mesh_sphere_radius)
+				range >	m_control.wanted_range * BS + mesh_sphere_radius)
 				continue; // Out of range, skip.
 
 			// Keep the block alive as long as it is in range.
 			block->resetUsageTimer();
 			blocks_in_range_with_mesh++;
 		}
 	}
+
+	m_control.loaded_range = loaded_range / BS;
```

Comment:
what's this used for?

Reply:
This is the limit for "unlimited view range" in updateDrawList

Reply:
Is the reason for this limit that the culler can't possibly operate on the entire map?
or could we just pass 65536 to `getBlocksInViewRange`?

Reply:
This is needed to limit the search-space, otherwise we'd recurse unnecessarily through many un-loaded (and assumed-transparent) blocks.

@x2048 Now that I look at the code again... A simple style nit is to always pass the range to `getBlocksInViewRange` and remove the -1.0f default and handling. The it could even be made static (if only to indicate that it does not access any members.)

Reply:
I'm wondering: If the culler does not traverse the entire world how does it know where the farthest-out block is?

In any case this does actually work in practice...

Reply:
The culler traverses up to the loaded_range which is calculated from the blocks the are loaded

Reply:
Traversing non-existing blocks infinitely far is not ok.
I've tried teleporting ca. 20000 nodes away with full view range enabled, and the game (as to expect) got completely hung up.

Possible solutions:
* Don't do any  occlusion checks at all if full view range is enabled, and only iterate through all existing blocks.
* Like above, but do occlusion checks for nearby blocks (for better performance in many situations).
* Do (new-style) occlusion testing per set of existing connected blocks: First, find all these sets of connected blocks (i.e. by traversing all existing blocks). Then, for each of these sets, start by considering all blocks that have non-existing blocks next to them towards the camera direction, and do the usual traversal from there (you might have to traverse sorted by distance now though, as blocks are no longer inserted in distance-sorted order). Don't traverse non-existing blocks. If full view range is disabled, do traverse non-existing blocks, as the end condition is view distance (which is small), not exhaustiveness of visited existing blocks.